### PR TITLE
Add new-repo openshift/image-customization-controller

### PR DESCRIPTION
**Pull request by asalkeld**
/cc @zaneb @kirankt

### CODE REVIEW

File: ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.9"
+  namespace: ocp
+tests:
+- as: e2e-aws
```
Comment (andfasano) [inline ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml:20]: Why `e2e-aws` and not our usual metal-ipi tests?

Comment (asalkeld) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml:20]: Hi @andfasano 
So I am not very clued up on these jobs.. But I ran "make new-repo" and it added it.
Also I looked at  cbo: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-baremetal-operator/openshift-cluster-baremetal-operator-master.yaml#L128-L131
and it does not have an aws job in CI

Comment (zaneb) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml:20]: Not sure why that is there in the CBO, but given that it is a CVO Operator (i.e. can block the rollout completing) maybe it's there to make sure we don't screw up other platforms... which is a thing that has happened iirc.
I don't think the same applies here, so there's reason to run an AWS job for this repo imho. I get why it's in the default set, but this repo is very platform-specific.

Comment (asalkeld) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml]: I'll add unit and lint tests for now (no point in e2e until we have images and we are running the controller

Comment (asalkeld) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml]: we will need to merge this first to get the makefile https://github.com/openshift/image-customization-controller/pull/2

Comment (andfasano) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml]: "make new repo" => fine didn't realize it :) we could add all the missing baremetal required tests later. I think it's the reasonable default for creating a new OpenShift component repo and yes, we kept it in CBO to be sure to not screw up other platforms, but if it's not strictly required here then we could remove it to save some resources.

/lgtm

Comment (andfasano) [reply ci-operator/config/openshift/image-customization-controller/openshift-image-customization-controller-main.yaml]: /hold still some config to be fixed

**Pull request merged**